srcDLL/DllMain.cpp

```diff
@@ -76,7 +79,23 @@ BOOL WINAPI DllMain(HINSTANCE hInstance, DWORD dwReason, LPVOID reserved)
 	return TRUE;
 }
 
-int TApp::Init()
+bool InstallTAppEventHooks()
+{
+	// Replace call to gTApp.Init with custom routine
+	if (!Op2RelinkCall(0x004A8877, GetMethodVoidPointer(&TApp::Init))) {
+		return false;
+	}
+
+	// Replace call to gTApp.ShutDown with custom routine
+	Op2RelinkCall(0x004A88A5, GetMethodVoidPointer(&TApp::ShutDown));
+
+	// Replace call to LoadLibrary with custom routine (address is indirect)
+	Op2MemSetDword(loadLibraryDataAddr, &loadLibraryNewAddr);
+
+	return true;
+}
+
+bool InstallDepPatch()
```

Comment:
I don't really know what a DEP is or the DSEG segment. Might be worth explaining a bit more in comments?

---

It might be wise to put a declaration for the function InstallDepPatch at the top of the file in case it gets moved around. It would follow the current convention a bit better I think.